src/rules.ts

```diff
@@ -25,14 +25,16 @@ export const rules: TSESLint.RuleModule<MessageIds, [Options]> = createRule<
 		const ruleBans = rules.map(({ message, rule }) => ({
 			message,
 			rule,
-			tester: new RegExp(`eslint-disable(?:-next-line)?.*${rule}`),
+			tester: new RegExp(
+				`^eslint-disable(?:-next-line)?\\s+.*(?<=,|\\s)${rule}(?=,|\\s|$)`,
```

Comment:
Additions:
1. single or more spaces after the `eslint-disable` comment, to avoid catching invalid comments
2. positive lookbehind for either a space or a comma, to ensure that this is the start of the rule name in the comment (whether it's a single rule or multiple comma-separated rules)
3. positive lookahead for either a space or comma, to ensure that this is the end of the rue name (same a sbove)

here is a [regex101](https://regex101.com/r/0cBcm5/1)

Reply:
Great explanation 🙂 thanks!